Notify/pushjet/pushjet.py

```diff
@@ -0,0 +1,325 @@
+        as seconds from epoch.
+    :ivar device_uuid: The UUID of the device that owns the subscription.
+    """
+
+    def __repr__(self):
+        return "<Pushjet Subscription to service \"{}\">".format(repr_format(self.service.name))
+
+    def __init__(self, subscription_dict):
+        self.service = Service._from_data(subscription_dict['service'])
+        self.time_subscribed = subscription_dict['timestamp']
+        self.last_checked = subscription_dict['timestamp_checked']
+        self.device_uuid = subscription_dict['uuid'] # Not sure this is needed, but...
+
+class Message(object):
+    """A message received from a service.
+    
+    :ivar message: The message body.
+    :ivar title: The message title. May be ``None``.
+    :ivar link: The URL the message links to. May be ``None``.
+    :ivar time_sent: When the message was sent, as seconds from epoch.
+    :ivar importance: The message's priority level between 1 and 5, where 1 is
+        least important and 5 is most.
+    :ivar service: The :class:`~pushjet.Service` that sent the message.
+    """
+
+    def __repr__(self):
+        return "<Pushjet Message: \"{}\">".format(repr_format(self.title or self.message))
+
+    def __init__(self, message_dict):
+        self.message = message_dict['message']
+        self.title = message_dict['title'] or None
+        self.link = message_dict['link'] or None
+        self.time_sent = message_dict['timestamp']
+        self.importance = message_dict['level']
+        self.service = Service._from_data(message_dict['service'])
+
+class Api(object):
+    """An API with a custom URL. Use this if you're connecting to a self-hosted
+    Pushjet API instance, or a non-standard one in general.
+
+    :param url: The URL to the API instance.
+    :ivar url: The URL to the API instance, as supplied.
+    """
+
+    def __repr__(self):
+        return "<Pushjet Api: {}>".format(self.url).encode(sys.stdout.encoding, errors='replace')
+
+    def __init__(self, url):
+        self.url = unicode_type(url)
+        self.Service = with_api_bound(Service, self)
+        self.Device = with_api_bound(Device, self)
+    
+    def _request(self, endpoint, method, params=None, data=None):
+        url = urljoin(self.url, endpoint)
+        try:
+            r = requests.request(method, url, params=params, data=data)
+        except requests.RequestException as e:
+            raise RequestError(e)
+        status = r.status_code
+        if status == requests.codes.INTERNAL_SERVER_ERROR:
+            raise ServerError(
+                "An error occurred in the server while processing your request. "
+                "This should probably be reported to: "
+                "https://github.com/Pushjet/Pushjet-Server-Api/issues"
+            )
+        try:
+            response = r.json()
+        except ValueError:
+            response = {}
+        return status, response
+
+default_api = Api(DEFAULT_API_URL)
```

Comment:
Just curious... this code gets initialized every time your code is sourced (even for those not using this plugin).  Is this necessary?

I was also curious, i noticed the content in _Notify/pushjet_ has no license associated with it.  Is this your code? Or is there a source (to credit)?